### PR TITLE
Styleguide

### PR DESCRIPTION
**Pull request by cdaringe**
# problem statement

# solution

# discussion

### CODE REVIEW

**Comment by cdaringe** [comment]
**[Storybook Hub](https://hub.getstorybook.io/)** is building storybooks for every commit in this PR.

**Latest Storybook: https://5cd74366-d572-4a2d-8202-558bc4b7ff17.sbook.io/**
<sub>( Visit [here](https://hub.getstorybook.io/app/github.com/Tripwire/octagon/branch/styleguide/) for more info )</sub>

> _This comment is automatically added by Storybook Hub_

**Comment by cdaringe** [comment]
storybook-merge

**Comment by jhegg** [comment]
Nice, that's very helpful.

File: scripts/suir-story-util.js [hunk 1]
```diff
@@ -9,15 +9,19 @@ const naturalSort = require('javascript-natural-sort')
...
-//   .then(suir.mapToURIs)
-//   .then(suir.mapToIFrames)
-//   .then(suir.mapToStories)
-//   .then(suir.concatImportsAndStories)
-//   .then(txt => suir.writeStories(this.suirStoriesDest))
-
+/**
+ * @module suir-story-util
+ * @description utility class to auto-generate SUIR stories.  it enables us
+ * to parse our own suir imported components and generate a "story" for for
```
Comment (jhegg) [inline scripts/suir-story-util.js]: s/for for/for

**Comment by cdaringe** [comment]
@eddier,
- i accidentally trashed a few stories that were non-SUIR related from the stories dir. ill take a look at that soon
- the last thing i want to do is inject our CSS into SUIR iframe.  then, no more SUIR stories necessary

**Comment by cdaringe** [comment]
untitled mov

alright, new strategy is in.  `styleguidist` is now a first class citizen, storybook is a 2nd class citizen.  in dev, just use styleguidist (`yarn start`).  storybook still delivers the cloud build, so lets continue w/ it for that functionality only.

File: .storybook/config.js [hunk 2]
```diff
@@ -1,22 +1,13 @@
 import '../lib/styles/semantic.css'
 import './styles.css'
 import { configure } from '@kadira/storybook'
-import { basename } from 'path'
-const naturalSort = require('javascript-natural-sort')
```
Comment (cdaringe) [inline .storybook/config.js]: 🔥 drop normal story dirs
🍏 add styleguist story

File: .storybook/styleguidist/styleguidist.stories.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import React from 'react'
+import { storiesOf } from '@kadira/storybook'
+
+const sgURI = process.env.NODE_ENV.match(/devel/) ? 'http://localhost:3000' : '/styleguide'
+console.log(`env: ${process.env.NODE_ENV}`)
+console.log(`sourcing styleguide from: ${sgURI}`)
+export default storiesOf('components', module)
+.add('via styleguidist', () => <iframe src={sgURI} width='100%' height='99%' />)
```
Comment (cdaringe) [inline .storybook/styleguidist/styleguidist.stories.js]: iframe into static styleguidist build

File: semantic.json [hunk 4]
```diff
@@ -19,5 +19,5 @@
   "permission": false,
   "autoInstall": false,
   "rtl": false,
-  "version": "2.2.7"
+  "version": "2.2.9"
```
Comment (cdaringe) [inline semantic.json]: oh snap, turns out the `semantic-ui` package has some smarts in it that updates our theme!  a yarn.lock conflict led me to purge the lockfile, and it updated the theme (patch only)

File: src/components/FavoriteButton.examples.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
+const FavoriteButton = require('../src/components/FavoriteButton');
```
Comment (cdaringe) [inline src/components/FavoriteButton.examples.md]: wooohhoooo, **no more \*.stories.js** files. all stories are styleguidist friendly

**Comment by cdaringe** [comment]
this PR is huuuge, so lets sit down together for a review!

**Pull request merged**